internal/internal_workflow.go

```diff
@@ -664,6 +664,24 @@ func assertNotInReadOnlyState(ctx Context) {
 	}
 }
 
+func assertNotInReadOnlyStateCancellation(ctx Context) {
```

Comment:
Is this "read only" the same definition as used for query and update validators?

Reply:
No, as is documented in the function we can't use the normal check in cancellation because of how the Go SDK implemented workflow cancellation

Reply:
Hrmm, I wonder if there's another term for read-only to not be confused with the query/update-validator definition. Either way, just naming, no biggie.

Reply:
I named them similar because it is basically the same validation for the same purpose.